### PR TITLE
possibility to use a python variable file for linting

### PR DESCRIPTION
**Pull request by viktor1230**
Correction according to issues: #953 #1031 #1005

### CODE REVIEW

File: robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py
Comment (fabioz) [inline robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py:16]: The files under `robocorp_ls_core/libs/robocop_lib` are vendored files from `robocop` (so, they shouldn't really be used nor imported inside of the language server aside from just from its main public API to collect errors).

See the readme under `robocorp_ls_core/libs/robocop_lib` if an update is needed on how to update it (but if it has to be updated, it should be in its own PR).

File: robotframework-ls/src/robotframework_ls/impl/variales_from_variable_file.py
Comment (fabioz) [inline robotframework-ls/src/robotframework_ls/impl/variales_from_variable_file.py:9]: There should be at least one new test case for this one.

See: robotframework_ls_tests.test_variables_from_arguments_file.test_variables_from_arguments_file as a base on how it should look like.

File: robotframework-ls/package.json
Comment (fabioz) [inline robotframework-ls/package.json:444]: There should be at least one test-case for this setting.

See robotframework_ls_tests.test_vscode_robot.test_variables_completions_integrated_using_config as a base.

**Review by fabioz — CHANGES_REQUESTED** [review]
Thank you for the PR! Just a few notes added ;)

**Comment by viktor1230** [comment]
Realized i messed up argument files, will try to correct according to this and your comments. Thanks.

File: robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py
Comment (viktor1230) [reply robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py:16]: Thank you for your comments. I have corrected the other ones but I don't understand this one as I am not familiar with vendored files. Should this not be included here?

Comment (fabioz) [reply robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py:16]: Well, the files in `robocorp_ls_core/libs/robocop_lib` are actually from the 'https://pypi.org/project/robotframework-robocop/` project, so, we should not do any manual change on those files. 

If they need to be updated you need to follow the instructions from: https://github.com/robocorp/robotframework-lsp/blob/master/robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/README.md (but as I see no updates in the `.dist-info` with a version update, I believe this wasn't done...).

Comment (fabioz) [reply robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py:16]: As a note, any changes under `https://github.com/robocorp/robotframework-lsp/blob/master/robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/` must be reverted in this PR...

Comment (fabioz) [reply robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py:16]: If we need something from them, we should ask the project maintainers to update it and then we can update the vendored files.

Comment (viktor1230) [reply robocorp-python-ls-core/src/robocorp_ls_core/libs/robocop_lib/robocop/config.py]: Thanks for your explanation. I reverted the changes.

**Comment by fabioz** [comment]
Waiting for the tests to be fixed... there are typing issues with mypy and `"ImportError: cannot import name 'OPTION_ROBOT_VARIABLES_LOAD_FROM_VARIABLES_FILE' from 'robotframework_ls.impl.robot_lsp_constants'`...

**Comment by viktor1230** [comment]
I'm sorry it took some time, but now I have attempted to fix according to tests.

**Comment by viktor1230** [comment]
I believe I have fixed some of the failing tests. However there is still a few left that I don't really can get my head around, test_rcc_cloud in robocorp for example. How may my changes have affected that?

**Comment by viktor1230** [comment]
@fabioz Do you mind triggering the tests again?

**Comment by viktor1230** [comment]
Are the fails that are left related to my changes?

File: robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py
Comment (fabioz) [inline robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py:32]: Is this correct? Can't the `#` be escaped? Doesn't Robot Framework itself have some API which could be used to read this file?

i.e.: I think the current parsing can be pretty brittle (I'm ok in merging as is though if you don't want to pursue it further).

**Comment by fabioz** [comment]
> Are the fails that are left related to my changes?

No, I don't think so (just waiting for answer to my previous comment to decide whether to merge as is or not).

File: robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py
Comment (viktor1230) [reply robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py:32]: I have looked around without finding an API which can do the parsing. I noticed that comments in multiline variable, dict etc. screwed things up. As far as i have tested and used it this works for now.

Comment (fabioz) [reply robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py:32]: I think this is actually a regular python file.

You should be able to use: `robotframework_ls.impl.variable_completions_from_py.collect_variables_from_python_ast` to collect the variables from it.

Comment (viktor1230) [reply robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py:32]: How can I call that function when I have path as variable?

Comment (viktor1230) [reply robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py:32]: @fabioz Should i create a RobotDocument of the file to be able to parse it with collect_variables_from_python_ast?

Comment (viktor1230) [reply robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py]: I think I have corrected according to comments now. Could you please retrigger tests?

Comment (viktor1230) [reply robotframework-ls/src/robotframework_ls/impl/variables_from_variable_file.py]: @fabioz I think it might pass the tests now, do you mind triggering it?

**Comment by viktor1230** [comment]
@fabioz I'm sorry for bugging you. But i think the tests might pass now, if you could trigger them?

**Comment by fabioz** [comment]
Thank you for the PR!

This is now merged in https://github.com/robocorp/robotframework-lsp/commit/afda578e5293fbc823e2894eef31e5ccad9acfa6

Note: I did fix the `test_variables_from_variables_file` which was broken and made an additional commit to fix the code generation (https://github.com/robocorp/robotframework-lsp/commit/51dbf3755cf7cda2f063cf33115a038e4c2bffb7).

I'll create a new release with this shortly.

**Pull request closed without merging**
